test/Microsoft.AspNet.Mvc.IntegrationTests/BinderTypeBasedModelBinderIntegrationTest.cs

```diff
@@ -0,0 +1,305 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNet.Mvc.ModelBinding;
+using Xunit;
+
+namespace Microsoft.AspNet.Mvc.IntegrationTests
+{
+    public class BinderTypeBasedModelBinderIntegrationTest
+    {
+        [Theory(Skip = "ModelBindingResult should be non null if a model binder returns a non null result.")]
+        [InlineData(typeof(NullModelBinder), true)]
+        [InlineData(typeof(NullModelNotSetModelBinder), false)]
+        public async Task BindParameter_WithModelBinderType_NoData(Type modelBinderType, bool isModelSet)
+        {
+            // Arrange
+            var argumentBinder = ModelBindingTestHelper.GetArgumentBinder();
+            var parameter = new ParameterDescriptor()
+            {
+                Name = "Parameter1",
+                BindingInfo = new BindingInfo()
+                {
+                    BinderType = modelBinderType
+                },
+
+                ParameterType = typeof(string)
+            };
+
+            // No data is passed.
+            var operationContext = ModelBindingTestHelper.GetOperationBindingContext(request => { });
+            var modelState = new ModelStateDictionary();
+
+            // Act
+            var modelBindingResult = await argumentBinder.BindModelAsync(parameter, modelState, operationContext);
+
+            // Assert
+
+            // ModelBindingResult
+            Assert.NotNull(modelBindingResult);
+            Assert.Null(modelBindingResult.Model);
+            Assert.Equal(isModelSet, modelBindingResult.IsModelSet);
+
+            // ModelState
+            Assert.True(modelState.IsValid);
+            var key = Assert.Single(modelState.Keys);
+            Assert.Equal("CustomParameter", key);
+            Assert.Empty(modelState[key].Errors);
```

Comment:
why just this key? Would be nice to see a verification for `ErrorCount`. I know that it's provably the same, it would be nice to be consistent in what these verification steps look like.